cmd/mungedocs/links.go

```diff
@@ -29,6 +29,8 @@ import (
 var (
 	// Finds markdown links of the form [foo](bar "alt-text").
 	linkRE = regexp.MustCompile(`\[([^]]*)\]\(([^)]*)\)`)
+	// Finds markdown link typos of the form (foo)[bar]
+	badLinkRE = regexp.MustCompile(`\(([^]()]*)\)\[([^]()]*)\]`)
```

Comment:
`FindString` and `MatchString` have nearly identical performance, so I removed the capture groups and switched to `FindString`. I think it's cleaner to keep this logic in links.go, since this data seems irrelevant to other mungers.